sql/core/src/test/scala/org/apache/spark/sql/streaming/FileStreamSourceSuite.scala

```diff
@@ -341,6 +347,33 @@ class FileStreamSourceSuite extends FileStreamSourceTest with SharedSQLContext {
     Utils.deleteRecursively(tmp)
   }
 
+  test("metadataPath should be in checkpointLocation") {
```

Comment:
What are you really testing here?  That its not just blindly ignoring the parameter that is passed to it?  Given the amount of reflection you are adding here it seems likely that the cost of maintaining this test outweighs its utility.